### DISCUSSION TITLE
Bugs/Devs = 12 years

### DISCUSSION THREAD

**Discussion by MhAllan** [dotnet/maui#15958] [General]
Hi,

We have 2400 bugs, and 4 devs. If the dev can fix a bug in one week then we need 2400/4 = 600 weeks. that is 12 years. 

I believe we need like 100 engineer here. I see booking.com have 2700 engineer, and I bet they did gofundme. Let's do like them especially Microsoft is only 2.5 trillion worth.

**Comment by MartyIX** [comment]
Please compute it also for 

* https://bugs.chromium.org/p/chromium/issues/list
* https://github.com/dotnet/roslyn/issues
* https://github.com/dotnet/runtime/pulls
* https://github.com/AvaloniaUI/Avalonia/issues

;)

-> IMHO it's about fixing *important* bugs. It's important if people can fix their issues as well. It's not the number of issues that's important. One can improve things forever.

**Reply by JRosanowski** [reply]
Sure, that's true. But I think people are losing confidence in the Maui Team because of how things are being managed. There's a whole bunch of important bugs that aren't being fixed in Maui .Net 7 and everyone is being told to wait until November for Maui .Net 8. In the meantime everyone is searching for workarounds or telling management that they can't do anything. Also Maui .Net 8 preview can't be used in production code and it has it's own issues. So we're all stuck until November and we've no indication that things will be better at that point. Will there be a bunch of show stopper bugs in .Net 8 and no indication of when they will be fixed?

**Reply by MartyIX** [reply]
I agree with you and I feel the same. 

The best thing is to actually test .NET 8 previews because they are the best shots right now. MAUI .NET 7 is not usable for non-trivial apps right now :|

**Reply by MhAllan** [reply]
@MartyIX it is not about bugs, it is we consumed all possible ways to communicate with this team that they need much bigger size.

**Reply by MartyIX** [reply]
> Let's do like them especially Microsoft is only 2.5 trillion worth.

This can intepreted also as "you are telling a 2.5 trillion company what to do". :) But joking aside, do you really believe that the MAUI team does not know it that a bigger team would likely do more work? At this point there are countless messages saying so. However, the way to phrase the question should imho be: *What would it take to hire more people?* 

Maybe there are not that many people who want to work on MAUI there are opportunities everywhere to do interesting stuff. Maybe there are architecture problems that needs to be ironned out before it makes even sense to hire more people, etc. So you need to first understand why they do nnot want to hire more people before telling them "hire more people". 

By the way, not every time havin more people means faster progress. Because of skill acquisition and because the people have to communicate.

**Reply by Hooterr** [reply]
@MartyIX That's the problem we hear nothing about from Microsoft. We don't know if they don't want more people working on it, can't find any people, or think it won't benefit the project to have more people. There's just dead silence about all those issues. If they don't provide official reasons for the state of MAUI then they force us to make guesses why things are the way they are. And my guess is that this project is a total dumpster and no one wants to come to work on it from inside or outside of Microsoft.

If you look at the devs working on it, you will quickly see that those are the same handful of people working on it for years since Xamarin.Forms days.

**Reply by MartyIX** [reply]
@Hooterr Would be great to ask on a call with the MAUI team or maybe at a Build event. But I hear you.

**Reply by MhAllan** [reply]
@MartyIX regarding "you are telling a 2.5 trillion company what to do" the answer is yes. like we told them IE is dying and they didn't get it. or that ASP webforms is bad, Goolges angular is bad, the whole world which is built on Html, Css, Js chose bad stack .. being 2.5 trillion company doesn't mean we don't have opinions and can't point to our observations. like this: https://github.com/dotnet/maui/discussions/15318#discussioncomment-6493229

**Comment by Mistofs** [comment]
Anyone walking in seeing this many open issues is unlikely to feel at ease about a product

**Reply by Hooterr** [reply]
From the project, we're working on for a customer in .net7, one of the first pages of the app:
Grid with a collection view. BUGGED. Need a workaround with a ContentView that serves the purpose of measuring space for collection view.
CollectionView with SwipeView. BUGGED. Need to write your own SwipeView because the MAUI one can't work properly with scrolling of the collection view.
Added a search bar. Disabled TextPrediction. BUGGED. Not even implemented on iOS.
Added Focused event to the search bar. BUGGED. Not even implemented on iOS.
Added another page to navigate to with Shell. BUGGED. Page title from the previous page stays on the top until the page finishes animating.
Added tinted image behavior from CT. TintColor BUGGED when you use binding with a converter.

Tried to do an early release for the client. Each developer got a different error while publishing the same code.

After such experience, you can't help but to laugh out loud.


Honestly the approach taken by Microsoft makes me want MAUI to fail and for Avalonia to take its place. At this point, I'd rather donate my free time by contributing to Avalonia than MAUI. Way to go Microsoft.

**Reply by MartyIX** [reply]
@Hooterr Similar experience here (eg https://github.com/dotnet/maui/pull/13814). Though it gets slowly better. However, have you guys considered contributing a patch? I'm just curious. I find it not very easy but for minor bugs it seems doable. For layout stuff it's offputting really.

**Reply by Xyncgas** [reply]
Why try opening a issue if there's ~3k unsolved, it's not likely getting fixed this year, at least we can talk about it and warn the others about this framework's reputation for them to look out for : There's a lot things you can't do right now using this framework, if a framework doesn't immediately solve your problem, find another one because you are not going to help microsoft develop their framework as someone who's got money to make and people to feed.

To be honest, there's a lot things you can do in this framework. My strategy is always what's the features in the framework that are stable, let's only use those so my expectation becomes what can I do with these it better be a lot, so I can build my library on top of these stable things let me be certain if there's a bug I can fix the bug.

All I've advocated, is a minimal framework with robust stable features, for me to be able to do everything (Events, Graphical, utilities). Let me get the ability to notify UI StateHasChanged like inside blazor, let me put a rectangle on screen using position absolute / relative and top left width height, I can get a grid for myself so I don't have to use the broken one in MAUI now

**Reply by Hooterr** [reply]
I'm not even complaining about the number of features
I'm not even complaining about the documentation 
I'm not even complaining about the architecture preference: MVVM over MVU
I'm not even complaining about the tooling
I'm not even complaining about different platform support

I'm complaining that when I want to make the simplest page consisting of a Grid, CollectionView and a Label work I need tons of workarounds because out of the box it's all broken. Grid overflows the screen, CollectionView has scrolling bugs unless you put HeightRequest on it and the Label if split into multiple lines overflows in auto height row....

Just yesterday I found out that on iOS in release mode, BackgroundColor on a Label just doesn't work. BACKGROUNDCOLOR! The simplest of things!

At this point, I don't know if I would even risk putting any of the MAUI code even close to production. This is bad.

This framework is in early beta. Don't let the GA label fool you.
